File: Visual Basic/PeopleWorkVB/PeopleWorkVB/Student.vb
' Class Student
' Damian Zylski
' 08/20/2020
Public Class Student
    Inherits Person

    'Attributes
    Private studentID As Integer
    Private major As String
    Private level As String

    'Constructors
    Public Sub New()
        MyBase.New()
        studentID = 0
        major = "Blank"
        level = "Blank"
    End Sub

    Public Sub New(n As String, id As Integer, m As String, l As String)
        MyBase.New(n)
        studentID = id
        major = m
        level = l
    End Sub

    'Setters
    Public Sub setID(id As Integer)
        studentID = id
    End Sub
    Public Sub setMajor(m As String)
        major = m
    End Sub
    Public Sub setLevel(l As String)
        level = l
    End Sub
    'Getters
    Public Function getID() As Integer
        Return studentID
    End Function
    Public Function getMajor() As String
        Return major
    End Function
    Public Function getLevel() As String
        Return level
    End Function

    'Output
    Public Overloads Sub writeOutput()
        Console.WriteLine("Name: " & MyBase.getName())
        Console.WriteLine("Student ID: " & studentID)
        Console.WriteLine("Major: " & major)
        Console.WriteLine("Level: " & level)
    End Sub
    'Equal
    Public Overloads Function equal(s As Student) As Boolean
        If (Me.studentID.Equals(s.studentID)) Then
            Return True
        Else
            Return False
        End If
    End Function

    'CompareTo
    Public Overloads Function compareTo(s As Student) As Integer
        If (Me.studentID > s.studentID) Then
            Return 1
        ElseIf (Me.studentID < s.studentID) Then
            Return -1
        ElseIf (Me.studentID = s.studentID) Then
            Return 0
        Else
            Return 0
        End If
    End Function

    Public Property gpa As Double

End Class

File: Java/LinkedLists/src/h3/Student.java
package h3;
//Student Class
//Damian Zylski
//11/15/2020

public class Student
{
    //attributes
    private String name;
    private int age;
    
    //constructor
    public Student()
    {
        name = "Blank";
        age = 0;
    }
    public Student(String n, int a)
    {
        name = n;
        age = a;
    }
    
    //getters
    public String getName()
    {
        return name;
    }
    public int getAge()
    {
        return age;
    }
    
    //setters
    public void setName(String n)
    {
        name = n;
    }
    public void setAge(int a)
    {
        age = a;
    }
    
    //write output
    public void writeOutput()
    {
        System.out.println("Name: " + name);
        System.out.println("Age: " + age);
        System.out.println("");
    }
    
    //equal
    public boolean equal(Student s)
    {
        if(this.getAge() == s.getAge())
        {
            return true;
        }
        else
        {
            return false;
        }
    }
    
    //compareTo
    public int compareTo(Student s)
    {
        if(this.getAge() > s.getAge())
        {
            return 1;
        }
        else if(this.getAge() < s.getAge())
        {
            return -1;
        }
        else
        {
            return 0;
        }
    }
    
    
    //toString
    @Override
    public String toString()
    {
        return "" + age;
    }
}
